src/frontend/screenshots.ts

```diff
@@ -0,0 +1,141 @@
+  const screenshotsDir =
+    process.env["SCREENSHOTS_DIR"] ?? "./screenshots/custom";
+  if (!existsSync(screenshotsDir)) {
+    mkdirSync(screenshotsDir, { recursive: true });
+  }
+
+  // Iterate the pages and screenshot them
+  for (const pageName of pageNames) {
+    // Skip the loader, because it's animated
+    if (pageName === "loader") {
+      continue;
+    }
+
+    // In the case of the faq we modify the URL slightly to show an open entry
+    if (pageName === "faq") {
+      await visit(browser, `http://localhost:8080/${pageName}#lost-device`);
+    } else {
+      await visit(browser, `http://localhost:8080/${pageName}`);
+    }
+
+    await browser.execute('document.body.style.caretColor = "transparent"');
+    await browser.saveScreenshot(`${screenshotsDir}/${pageName}.png`);
+  }
+}
+
+/** Create a chrome instance and run callback, deleting session afterwards */
+async function withChrome<T>(
+  cb: (browser: WebdriverIO.Browser) => T
+): Promise<T> {
+  // Screenshot image dimension, if specified
+  const { mobileEmulation } = readScreenshotsConfig();
+
+  const chromeOptions: ChromeOptions = {
+    args: ["headless", "disable-gpu"],
+    mobileEmulation,
+  };
+
+  const browser = await remote({
+    capabilities: {
+      browserName: "chrome",
+      "goog:chromeOptions": chromeOptions,
+    },
+  });
+
+  const res = await cb(browser);
+  await browser.deleteSession();
+  return res;
+}
+
+/** Visit page and wait until loaded */
+async function visit(browser: WebdriverIO.Browser, url: string) {
+  await browser.url(url);
+
+  /* Disable transitions to make sure we screenshot the (final) actual state */
+  await browser.execute(() => {
+    const notransition = `
+*, *::before, *::after {
+    -o-transition-property: none !important;
+    -moz-transition-property: none !important;
+    -ms-transition-property: none !important;
+    -webkit-transition-property: none !important;
+    transition-property: none !important;
+}
+        `;
+    const style = document.createElement("style");
+    style.appendChild(document.createTextNode(notransition));
+    document.body.appendChild(style);
+  });
+
+  /* Make sure everything has loaded */
+  await browser.waitUntil(
+    () => browser.execute(() => document.readyState === "complete"),
+    {
+      timeout: 10 * 1000,
+      timeoutMsg: "Browser did not load after 10 seconds",
+    }
+  );
+}
+
+/**
+ * Read the screenshots configuration based on 'SCREENSHOTS_TYPE'
+ * (either 'mobile' or 'desktop') and returns the appropriate device
+ * name and/or window size.
+ *
+ * NOTE: deviceMetrics are necessary due to a bug in webdriverio
+ * (otherwise just use 'deviceName'):
+ * * https://github.com/webdriverio/webdriverio/issues/8903
+ */
+function readScreenshotsConfig(): {
+  mobileEmulation?: {
+    deviceMetrics: { width: number; height: number };
+  };
+} {
+  const screenshotsType = process.env["SCREENSHOTS_TYPE"];
+  switch (screenshotsType) {
+    case "mobile":
```

Comment:
Yes, it's still necessary because the e2e tests use it. I'm not sure how much signal we get from the mobile tests, but they do tend to fail more often than the desktop ones, so _some_ signal I guess...